crates/phala-mq/src/dispatcher.rs

```diff
@@ -0,0 +1,156 @@
+use core::marker::PhantomData;
+
+use alloc::{collections::BTreeMap, vec::Vec};
+
+use crate::simple_mpsc::{channel, ReceiveError, Receiver as RawReceiver, Sender, Seq};
+use crate::types::{Message, Path};
+use crate::{BindTopic, MessageOrigin};
+use derive_more::Display;
+use parity_scale_codec::{Decode, Error as CodecError};
+
+impl Seq for (u64, Message) {
+    fn seq(&self) -> u64 {
+        self.0
+    }
+}
+
+#[derive(Default)]
+pub struct MessageDispatcher {
+    subscribers: BTreeMap<Path, Vec<Sender<(u64, Message)>>>,
+    sequence: u64,
```

Comment:
It means the dispatched event index in the current block. Maybe we can get a better name?

Reply:
done